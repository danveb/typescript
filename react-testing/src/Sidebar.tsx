export const Sidebar = ({ items }: {
  items: {
    id: number; 
    name: string; 
    href: string; 
  }[]; 
}) => (
  <div>
    {items.map((item) => (
      <div key={item.id}>
        <a role="navigation" href={item.href}>{item.name}</a>
      </div>
    ))}
  </div>
)