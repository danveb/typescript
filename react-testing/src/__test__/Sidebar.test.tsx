import { render, screen } from "@testing-library/react";
import { Sidebar } from "../Sidebar";

describe("test Sidebar component", () => {
  // smoke test
  test("renders without crashing", () => {
    const items = [
      {
        id: 1, 
        name: "test", 
        href: "/test", 
      }, 
    ]; 
    render(<Sidebar items={items} />); 
  });

  // snapshot test
  test("creates a snapshot", () => {
    const items = [
      {
        id: 1, 
        name: "test", 
        href: "/test", 
      }, 
    ]; 
    const { asFragment } = render(
      <Sidebar items={items} />
    ); 
    expect(asFragment()).toMatchSnapshot(); 
  }); 

  // // getAllByRole
  test("finds ariaRole navigation", () => {
    const items = [
      {
        id: 3, 
        name: "bye", 
        href: "/bye", 
      }, 
    ]; 
    render(<Sidebar items={items} />); 
    const anchorElements = screen.getAllByRole("navigation"); 
    expect(anchorElements[0]).toHaveTextContent(items[0].name); 
    expect(anchorElements[0]).toHaveAttribute("href", items[0].href); 
  }); 
});