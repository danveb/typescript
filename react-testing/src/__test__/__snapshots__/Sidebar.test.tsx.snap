// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Sidebar component creates a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <a
        href="/test"
        role="navigation"
      >
        test
      </a>
    </div>
  </div>
</DocumentFragment>
`;
